core/services/keeper/upkeep_executer.go

```diff
@@ -267,3 +280,14 @@ func addBuffer(val *big.Int, prct uint32) *big.Int {
 		100,
 	)
 }
+
+func (ex *UpkeepExecuter) turnBlockHashBinary(registry Registry, head *evmtypes.Head, lookback int64) (string, error) {
+	turnBlock := head.Number - (head.Number % int64(registry.BlockCountPerTurn)) - lookback
```

Comment:
might be worth adding something like...
```go
if turnBlock < 1 {
  logger.Error("misconfigured!")
  turnBlock = 1
}
```
... just to protect against mis-config on dev chains

Reply:
(we can address in a follow up if it becomes a recurring theme)